Mods/SML/Source/SML/Public/Patching/BlueprintHookHelper.h

```diff
@@ -30,24 +26,102 @@ class SML_API FBlueprintHookHelper {
+		fgcheckf(BaseProperty, TEXT("Did not find property %s on Context type %s"), PropertyName, *ContextClass->GetName());
+		T* Property = CastField<T>(BaseProperty);
+		fgcheckf(Property, TEXT("Property %s on Context type %s is not the requested type. Actual type: %s"), PropertyName, *ContextClass->GetName(), *BaseProperty->GetClass()->GetName());
+		return Property;
+	}
+
+public:
+	template<typename T>
+	typename T::TCppType* GetContextVarPtr(const TCHAR* PropertyName, int32 ArrayIndex = 0) const {
+		T* Property = GetContextProperty<T>(PropertyName);
+		return Property->GetPropertyValuePtr_InContainer(GetContext(), ArrayIndex);
+	}
+	
+	template<typename TEnum>
+	TEnum* GetContextVarEnumPtr(const TCHAR* PropertyName, int32 ArrayIndex = 0) const {
+		FEnumProperty* Property = GetContextProperty<FEnumProperty>(PropertyName);
+		// K2 only supports byte enums right now, according to a comment in EdGraphSchema_K2.cpp
+		return (TEnum*)Property->ContainerPtrToValuePtr<uint8>(GetContext(), ArrayIndex);
+	}
+	
+	// Bool properties have unique handling because they don't have a CppType because they can packed into
+	// a single bit and there's no CppType for that, so all reads/modifications have to go through the API
+	bool GetContextVarBool(const TCHAR* PropertyName, int32 ArrayIndex = 0) const {
+		FBoolProperty* Property = GetContextProperty<FBoolProperty>(PropertyName);
+		return Property->GetPropertyValue_InContainer(GetContext(), ArrayIndex);
+	}
+	
+	void SetContextVarBool(const TCHAR* PropertyName, bool Value, int32 ArrayIndex = 0) const {
+		FBoolProperty* Property = GetContextProperty<FBoolProperty>(PropertyName);
+		Property->SetPropertyValue_InContainer(GetContext(), Value, ArrayIndex);
+	}
+
 	/**
 	 * Retrieves local variable pointer from stack frame by the provided name
+	 * This method won't work on bool variables; use GetLocalBoolVar instead
 	 * This method won't work on [out] variables, as these are stored separately from local stack frame
 	 * Passed pointer is guaranteed to be valid for the duration of hook execution
 	 *
-	 * @param ParameterName name of the local variable to retrieve
+	 * @param VariableName name of the local variable to retrieve
 	 * @param ArrayIndex if variable represents a statically sized array, retrieve value at specified index
 	 * @return pointer to the local variable value
 	 * @note Will check false if provided variable name represents an [out] variable
 	 */
 	template<typename T>
-	typename T::TCppType* GetLocalVarPtr(const TCHAR* ParameterName, int32 ArrayIndex = 0) const {
-		T* Property = CastField<T>(FramePointer.Node->FindPropertyByName(ParameterName));
-		check(Property);
-		checkf(!Property->HasAnyPropertyFlags(CPF_OutParm), TEXT("Attempt to use GetLocalVarPtr on [out] variable"));
+	typename T::TCppType* GetLocalVarPtr(const TCHAR* VariableName, int32 ArrayIndex = 0) const {
+		T* Property = CastField<T>(FramePointer.Node->FindPropertyByName(VariableName));
+		fgcheck(Property);
```

Comment:
Could add a message to the check, like in `GetContextProperty`

Reply:
I will address this in a separate PR for the hook helper.